v2/api/dataprotection/customizations/backup_vaults_backup_instance_extensions.go

```diff
@@ -0,0 +1,158 @@
+	"github.com/Azure/azure-service-operator/v2/internal/set"
+	"github.com/Azure/azure-service-operator/v2/internal/util/to"
+	"github.com/Azure/azure-service-operator/v2/pkg/genruntime"
+	"github.com/Azure/azure-service-operator/v2/pkg/genruntime/extensions"
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+	"sigs.k8s.io/controller-runtime/pkg/conversion"
+)
+
+var _ extensions.PostReconciliationChecker = &BackupVaultsBackupInstanceExtension{}
+
+var nonRetryableStates = set.Make(
+	"ConfiguringProtectionFailed",
+	"Invalid",
+	"NotProtected",
+	"ProtectionConfigured",
+	"SoftDeleted",
+	"ProtectionStopped",
+	"BackupSchedulesSuspended",
+	"RetentionSchedulesSuspended",
+)
+
+const (
+	BackupInstancePollerResumeTokenAnnotation = "serviceoperator.azure.com/bi-poller-resume-token"
+)
+
+func GetPollerResumeToken(obj genruntime.MetaObject, log logr.Logger) (string, bool) {
+	log.V(Debug).Info("GetPollerResumeToken")
+	token, hasResumeToken := obj.GetAnnotations()[BackupInstancePollerResumeTokenAnnotation]
+	return token, hasResumeToken
+}
+
+func SetPollerResumeToken(obj genruntime.MetaObject, token string, log logr.Logger) {
+	log.V(Debug).Info("SetPollerResumeToken")
+	genruntime.AddAnnotation(obj, BackupInstancePollerResumeTokenAnnotation, token)
+}
+
+// ClearPollerResumeToken clears the poller resume token and ID annotations
+func ClearPollerResumeToken(obj genruntime.MetaObject, log logr.Logger) {
+	log.V(Debug).Info("ClearPollerResumeToken")
+	genruntime.RemoveAnnotation(obj, BackupInstancePollerResumeTokenAnnotation)
+}
+
+func (extension *BackupVaultsBackupInstanceExtension) PostReconcileCheck(
+	ctx context.Context,
+	obj genruntime.MetaObject,
+	owner genruntime.MetaObject,
+	resolver *resolver.Resolver,
+	armClient *genericarmclient.GenericClient,
+	log logr.Logger,
+	next extensions.PostReconcileCheckFunc,
+) (extensions.PostReconcileCheckResult, error) {
+	log.V(Debug).Info("Starting Post-reconcilation for Backup Instance")
+	backupInstance, ok := obj.(*dataprotection.BackupVaultsBackupInstance)
+	if !ok {
+		return extensions.PostReconcileCheckResult{},
+			errors.Errorf("cannot run on unknown resource type %T, expected *dataprotection.BackupVaultsBackupInstance", obj)
+	}
+
+	// Type assert that we are the hub type. This will fail to compile if
+	// the hub type has been changed but this extension has not
+	var _ conversion.Hub = backupInstance
+
+	if owner != nil &&
+		backupInstance != nil &&
+		backupInstance.Status.Id != nil &&
+		backupInstance.Status.Properties != nil &&
+		backupInstance.Status.Properties.ProtectionStatus != nil &&
+		backupInstance.Status.Properties.ProtectionStatus.Status != nil {
+
+		protectionStatus := *backupInstance.Status.Properties.ProtectionStatus.Status
+		log.V(Debug).Info(fmt.Sprintf("Protection Status is  %q", protectionStatus))
+
+		// return success for reconcilation if the state is non-retryable
+		if nonRetryableStates.Contains(protectionStatus) {
```

Comment:
Would suggest inversing this logic here. We can always return success, if` protectionStatus != "ProtectionError"`

Reply:
that way, we're not checking the list each time during the reconcile.